influxql/parser.go

```diff
@@ -1739,7 +1817,18 @@ func (p *Parser) parseUnaryExpr() (Expr, error) {
 }
 
 // parseRegex parses a regular expression.
-func (p *Parser) parseRegex() (Expr, error) {
+func (p *Parser) parseRegex() (*RegexLiteral, error) {
+	nextRune := p.peekRune()
+	if isWhitespace(nextRune) {
```

Comment:
`Parser.consumeWhitespace` is the function you're thinking of (probably) but it reads the next token and checks to see if it's `WS`.  If it's not, it puts it back in the token buffer...but not in the underlying scanner buffer.  It needs to be in the scanner so we can `peek` to see if the next char is the start of a regex (`/`).

We now use `/` as both a division operator and delimiters for regex tokens.  This makes it impossible for the scanner to know which it is.  Only the parser has enough context to know when a regex is expected.  That's why I was pushing so hard to use back-tick or any other unused symbol as the query delimiter instead of making an existing token dual-purpose.  But, now that it's done, `/` is nice.  :)